htdocs/xoops_lib/modules/protector/library/HTMLPurifier/DefinitionCache/Serializer.php

```diff
@@ -188,12 +198,13 @@ private function _write($file, $data, $config)
         if ($result !== false) {
             // set permissions of the new file (no execute)
             $chmod = $config->get('Cache.SerializerPermissions');
-            if (!$chmod) {
-                $chmod = 0644; // invalid config or simpletest
-            }
+            if ($chmod === null) {
+                // don't do anything
+            } else {
```

Comment:
__[phpmd]__
* The method _write uses an else expression. Else is never necessary and you can simplify the code to work without else.

<!-- Comment by [SideCI](https://sideci.com) -->